sky/provision/aws/instance.py

```diff
@@ -175,6 +188,24 @@ def cleanup_ports(
         'Values': [sg_name]
     }])
     if len(list(sgs)) != 1:
-        raise ValueError(f'Expected security group {sg_name} not found. '
-                         'Cannot cleanup ports.')
-    list(sgs)[0].delete()
+        logger.warning(f'Expected security group {sg_name} not found. '
+                       'Skip cleanup.')
+        return
+    backoff = common_utils.Backoff(initial_backoff=INITIAL_BACKOFF_SECONDS,
+                                   max_backoff_factor=MAX_BACKOFF_FACTOR)
+    for _ in range(MAX_ATTEMPTS):
+        try:
+            list(sgs)[0].delete()
+        except exceptions.ClientError as e:
+            match = re.search(_DEPENDENCY_VIOLATION_PATTERN, str(e))
+            if match is None:
+                raise e
+            if (match.group(1) == 'DependencyViolation' and
+                    match.group(2) == 'DeleteSecurityGroup'):
+                logger.warning(
+                    f'Security group {sg_name} is still in use. Retry.')
```

Comment:
```suggestion
                logger.debug(
                    f'Security group {sg_name} is still in use. Retry.')
```
Use debug to avoid polluting the progress bar.